lib/ui/window.dart

```diff
@@ -150,7 +150,37 @@ class Locale {
   /// Registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry).
   /// Typically this means the primary language subtag should be lowercase and
   /// the region subtag should be uppercase.
-  const Locale(this._languageCode, [ this._countryCode ]) : assert(_languageCode != null);
+  const Locale(
+    this._languageCode, [
+    this._countryCode,
+  ]) : assert(_languageCode != null),
+       this.scriptCode = null,
+       this._variants = null;
+
+  /// Creates a new Locale object.
+  ///
+  /// The keyword arguments specify the subtags of the Locale. The subtag values
+  /// should match the values listed as valid in Unicode CLDR supplemental data:
+  /// [language](http://unicode.org/cldr/latest/common/validity/language.xml),
+  /// [script](http://unicode.org/cldr/latest/common/validity/script.xml),
+  /// [region](http://unicode.org/cldr/latest/common/validity/region.xml) and
+  /// [variant](http://unicode.org/cldr/latest/common/validity/variant.xml). If
+  /// there is more than one variant, they should be in sorted order. This list
+  /// will be used as-is, and should never again be modified.
+  const Locale.create({
+    String language = 'und',
+    String script,
+    String region,
+    List<String> variants,
+  }) : assert(language == null || (language.length >= 2 && language.length <= 8
+                                   && language.length != 4)),
+       assert(script == null || script.length == 4),
+       assert(region == null || (region.length >= 2 && region.length <= 3)),
+       assert(variants == null || variants.length == 0 || variants[0].length >= 4),
```

Comment:
prefer_is_empty - but List.isEmpty isn't a const expression whereas "variants.length == 0" is. Shall I file a wishlist/bug report for Dart? :)

Reply:
Also "const_eval_type_bool_num_string" - I don't know why that one happened though.

Most tempting thing: removing the asserts. They "check syntax" in the sense of trying to assert some string-length requirements, but they don't check the contents of the strings. (Language and script must be just "a-z", region and variants can also contain digits.)